internal/config/config.go

```diff
@@ -830,6 +833,12 @@ func (c *Config) Validate(force bool) error {
 		}
 	}
 
+	if c.InternalLoadBalancer {
+		if c.GetProvider() != cloudprovider.AWS && c.GetProvider() != cloudprovider.GCP {
+			return &ValidationError{validationErrMsgs: []string{"internalLoadBalancer is only supported for AWS and GCP"}}
+		}
```

Comment:
Doesn't the `PrepareAzureControlPlane` function allow internal Loadbalancers on Azure?

Reply:
I forgot to mention the state of Azure. I now added a note in the PR description. In short: we need this function but also need to remove konnectivity. 

If you want to, we could remove (comment out?) the function until it is actually used.